### DISCUSSION TITLE
Issue with TypeError in ChemProp Hyperparameter Optimization

### DISCUSSION THREAD

**Discussion by TianYaoCSU** [chemprop/chemprop#1112] [Q&A]
I’m currently using ChemProp for hyperparameter optimization, and I encountered the following error during training:
TP = torch.cat(self.TP, dim=0).sum(0, keepdim=True)
TypeError: cat() received an invalid combination of arguments - got (Tensor, dim=int), but expected one of:
(tuple of Tensors tensors, int dim = 0, *, Tensor out = None)
(tuple of Tensors tensors, name dim, *, Tensor out = None)
This occurred while running the command:
css
chemprop hpopt -i F_reduced_labeled.csv -o hp_result -t classification --num-replicates 5 --metrics roc prc accuracy f1 binary-mcc --save-smiles-splits --show-individual-scores --pytorch-seed 42 --data-seed 42 --search-parameter-keywords depth ffn_num_layers dropout init_lr max_lr final_lr warmup_epochs batch_size --raytune-num-samples 50 --raytune-search-algorithm optuna --raytune-trial-scheduler AsyncHyperBand --raytune-num-cpus 8
Could you please help me understand what might be causing this issue? Any guidance would be greatly appreciated!
Thank you for your assistance!

**Comment by KnathanM** [comment]
Hi @TianYaoCSU, can you open this as an issue? The first few lines of your dataset (F_reduced_labeled.csv) would also be helpful. Can you try this more minimal example to see if it works? `chemprop train -i F_reduced_labeled.csv -t classification --metrics binary-mcc` This command works with the example classification on the github `tests/data/classification/mol.csv`.

**Comment by KnathanM** [comment]
This was turned into issue #1121
